controllers/cloud.redhat.com/providers/kafka/appinterface.go

```diff
@@ -69,9 +69,12 @@ func validateKafkaTopic(ctx context.Context, cl client.Client, nn types.Namespac
 	err := cl.Get(ctx, nn, &t)
 
 	if err != nil {
-		return &errors.MissingDependencies{
-			MissingDeps: map[string][]string{"topics": {nn.Name}},
-		}
+		missingDeps := errors.MakeMissingDependencies(errors.MissingDependency{
+			Source:  "kafka",
+			App:     nn.Name,
```

Comment:
I believe `nn.Name` is the name of the KafkaTopic we tried to fetch, not the app's name.

Reply:
You're right - but I'm not sure how to get the app name here. I'm looking at it now - I likely can get it from the ctx some how but I don't immediately know how to do it.

Reply:
I amended the method signature to accept the clowd app.